stdlib/LinearAlgebra/src/structuredbroadcast.jl

```diff
@@ -64,7 +64,24 @@ find_bidiagonal() = throw(ArgumentError("could not find Bidiagonal within broadc
 find_bidiagonal(a::Bidiagonal, rest...) = a
 find_bidiagonal(bc::Broadcast.Broadcasted, rest...) = find_bidiagonal(find_bidiagonal(bc.args...), rest...)
 find_bidiagonal(x, rest...) = find_bidiagonal(rest...)
+
+merge_uplos(::Nothing, ::Nothing) = nothing
+merge_uplos(a, ::Nothing) = a
+merge_uplos(::Nothing, b) = b
+merge_uplos(a, b) = a == b ? a : 'T'
+
+find_uplo(a::Bidiagonal) = a.uplo
+find_uplo(a) = nothing
+find_uplo(bc::Broadcasted) = mapreduce(find_uplo, merge_uplos, bc.args, init=nothing)
+
 function structured_broadcast_alloc(bc, ::Type{<:Bidiagonal}, ::Type{ElType}, n) where {ElType}
+    uplo = Nothing
```

Comment:
Now this can be greatly simplified, too! In fact, I don't think we need `find_bidiagonal` anymore at all — I think its definitions could even be deleted.  We can unconditionally `find_uplo(bc)` and use that answer directly in all cases.

Reply:
removed in 0d984ce !